### PR TITLE
statistics: add more write query kind

### LINKED ISSUE #10507: statistics: write qps in query stats is inconsistent with Prometheus
### What version of TiKV are you using?
master

### Steps to reproduce
sysbench test

### What did you expect?

consistent

### What did happened?
image

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: close #10507 

Problem Summary:

### What is changed and how it works?

it collect quert stats in scheduler and flush them in tick to pd.rs

What's Changed:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

insert 1200 tables
image

write test with Pessimistic

image

image

### Release note 

```release-note
None.
```

### CODE REVIEW

**Comment by nolouch** [comment]
cc @JmPotato

File: components/raftstore/src/store/worker/pd.rs [hunk 1]
```diff
@@ -873,13 +885,13 @@ where
             let read_bytes = region_peer.read_bytes - region_peer.last_store_report_read_bytes;
             let read_keys = region_peer.read_keys - region_peer.last_store_report_read_keys;
             let read_query_stats = region_peer
-                .read_query_stats
```
Comment (nolouch) [inline components/raftstore/src/store/worker/pd.rs:876]: it should including read and write stats?

File: components/raftstore/src/store/worker/pd.rs [hunk 2]
```diff
@@ -873,13 +885,13 @@ where
...
                 .sub_query_stats(&region_peer.last_store_report_query_stats);
             region_peer.last_store_report_read_bytes = region_peer.read_bytes;
             region_peer.last_store_report_read_keys = region_peer.read_keys;
             region_peer
                 .last_store_report_query_stats
-                .fill_query_stats(&region_peer.read_query_stats);
+                .fill_query_stats(&region_peer.query_stats);
             if read_bytes < hotspot_byte_report_threshold()
                 && read_keys < hotspot_key_report_threshold()
                 && read_query_stats.get_read_query_num() < hotspot_query_num_report_threshold()
```
Comment (nolouch) [inline components/raftstore/src/store/worker/pd.rs:897]: Does it still suitable?

**Review by nolouch — APPROVED** [review]
lgtm

**Review by 5kbpers — DISMISSED** [review]

File: components/raftstore/src/store/worker/pd.rs [hunk 3]
```diff
@@ -872,25 +884,25 @@ where
...
             region_peer.last_store_report_read_bytes = region_peer.read_bytes;
             region_peer.last_store_report_read_keys = region_peer.read_keys;
             region_peer
                 .last_store_report_query_stats
-                .fill_query_stats(&region_peer.read_query_stats);
+                .fill_query_stats(&region_peer.query_stats);
             if read_bytes < hotspot_byte_report_threshold()
                 && read_keys < hotspot_key_report_threshold()
-                && read_query_stats.get_read_query_num() < hotspot_query_num_report_threshold()
+                && query_stats.get_read_query_num() < hotspot_query_num_report_threshold()
```
Comment (5kbpers) [inline components/raftstore/src/store/worker/pd.rs:897]: Should we add a condition for write queries here?

Comment (lhy1024) [reply components/raftstore/src/store/worker/pd.rs:897]: Currently, the read hotspot information is obtained from the store heartbeat, and the write hotspot information is obtained from the region heartbeat, and the corresponding threshold will be added when all the hotspot information is done in the store heartbeat later.

File: src/read_pool.rs [hunk 4]
```diff
@@ -299,6 +299,7 @@ mod tests {
 
     impl FlowStatsReporter for DummyReporter {
         fn report_read_stats(&self, _read_stats: ReadStats) {}
+        fn report_write_stats(&self, _write_stats: WriteStats) {}
```
Comment (Yisaer) [inline src/read_pool.rs:302]: I noticed that `FlowStatsReporter` will report write stats while the file was still named as `read_pool.rs`.

**Review by Yisaer — APPROVED** [review]
basically lgtm

File: src/read_pool.rs (hunk 4 above)
Comment (lhy1024) [reply src/read_pool.rs:302]: I did not find a more suitable place, do you have any good suggestions

**Comment by lhy1024** [comment]
@5kbpers @Yisaer PTAL

**Review by nolouch — APPROVED** [review]

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**